### DISCUSSION TITLE
Setup Z-push for Zimbra

### DISCUSSION THREAD

**Discussion by cedrichatem** [Z-Hub/Z-Push#32] [General]
Hello All,

I found several documentations related to Z-Push and Zimbra but none of them seems uptodate or related to the setup I'm trying to do.
So I need you help ;-)

I've a Zimbra 8.8.15 server
I've build a Zpush server on witch I've installed Apache & Zpush 2.7.1.

And now comes the setup of the Backend....
Should I download additional Backend like : https://sourceforge.net/projects/zimbrabackend/ ?
Some docs since to tell that The zimbra backed is intergrated and just add thoses lines : 
    define('BACKEND_PROVIDER', 'BackendZimbra');
    define('ZIMBRA_URL', 'https://---my-zimbra-IP---');
Some docs give more parameters...
	define('ZIMBRA_DISABLE_URL_OVERRIDE', false);
	define('ZIMBRA_RETRIES_ON_HOST_CONNECT_ERROR',5);
	define('ZIMBRA_USER_DIR', 'zimbra');
	define('ZIMBRA_TIMEZONE', 'America/Sao_Paulo');
	define('ZIMBRA_VIRTUAL_APPOINTMENTS',true);
	define('ZIMBRA_DISABLE_MULTI_CALENDARS',true);
	define('ZIMBRA_SERVER_INVITE_REPLY', true);
	define('ZIMBRA_VIRTUAL_TASKS',true);
	define('ZIMBRA_DISABLE_MULTI_TASK_LISTS',true);
	define('ZIMBRA_DISABLE_TASKS',true);
	define('ZIMBRA_VIRTUAL_CONTACTS',true);
	define('ZIMBRA_IGNORE_EMAILED_CONTACTS',true);
	define('ZIMBRA_SYNC_CONTACT_PICTURES', true);
	define('ZIMBRA_DISABLE_NOTES',true);
	define('ZIMBRA_ENFORCE_VALID_EMAIL', true);
	define('ZIMBRA_HTML',true);
	define('ZIMBRA_SMART_FOLDERS',true);
	etc...

What should I do ?
Could you guide me ?

Thanks

Cédric

My logs : 
z-push-error.log
`27/10/2023 08:34:54 [ 1883] [FATAL] [ch] FatalMisconfigurationException: Backend provider 'BackendZimbra' can not be loaded. Check configuration! - code: 0 - file: /var/www/html/lib/core/zpush.php:642
27/10/2023 08:34:54 [ 1883] [FATAL] [ch] Exception: (FatalMisconfigurationException) - Backend provider 'BackendZimbra' can not be loaded. Check configuration!
`

z-push.log
`27/10/2023 08:35:49 [ 1884] [FATAL] [ch] FatalMisconfigurationException: Backend provider 'BackendZimbra' can not be loaded. Check configuration! - code: 0 - file: /var/www/html/lib/core/zpush.php:642
27/10/2023 08:35:49 [ 1884] [ INFO] [ch] User-agent: 'unknown'
27/10/2023 08:35:49 [ 1884] [FATAL] [ch] Exception: (FatalMisconfigurationException) - Backend provider 'BackendZimbra' can not be loaded. Check configuration!
27/10/2023 08:35:49 [ 1884] [ INFO] [ch] cmd='' memory='1.40 MiB/2.00 MiB' time='0.01s' devType='' devId='' getUser='myuser' from='172.16.16.16' idle='0s' version='2.7.1
' method='GET' httpcode='500'
`

**Comment by matidau** [comment]
Hiya,

Yes you will need to download the Zimbra Backend. 

All the ZIMBRA prefixed options such as ZIMBRA_URL should be edited in the file backend/zimbra/config.php

The Zimbra Backend is a separate project and you might be better served asking them directly for support  https://sourceforge.net/p/zimbrabackend/support-requests/

There is another project that has both Z-Push and the Zimbra Backend mostly preconfigured, but this uses Jetty as the web server. https://github.com/imanudin11/zcs-zpush

Hope that helps. 

When you get it up and running, feel free to add an entry to the wiki https://github.com/Z-Hub/Z-Push/wiki/Installation

Cheers, 
Mat
